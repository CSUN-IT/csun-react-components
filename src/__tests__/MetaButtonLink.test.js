import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaButtonLink from '../../src/components/meta/ButtonLink';

configure({ adapter: new Adapter() });

describe('MetaButtonLink', () => {
  it('Should be defined', () => {
    expect(MetaButtonLink).toBeDefined();
  });

  it('Renders component correctly', () => {
    const mockLink = '/docs';
    const tree = shallow(
      <MetaButtonLink text={'Button'} color={'default'} size={'sm'} to={mockLink} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
