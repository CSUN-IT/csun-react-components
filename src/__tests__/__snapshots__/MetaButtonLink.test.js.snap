// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaButtonLink Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaButtonLink
    btnStyle=""
    className=""
    color="default"
    size="sm"
    text="Button"
    to="/docs"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "ariaCurrent": "true",
      "children": "Button",
      "className": "btn btn-default btn-sm",
      "to": "/docs",
    },
    "ref": null,
    "rendered": "Button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "ariaCurrent": "true",
        "children": "Button",
        "className": "btn btn-default btn-sm",
        "to": "/docs",
      },
      "ref": null,
      "rendered": "Button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
